---
layout: default
---
## Common Bioinformatics Software

* **[StaPH-B](https://github.com/StaPH-B)** – State public health bioinformaticians workgroup

* **[ABiL(Applied Bioinformatics Laboratory)](https://www.abil.ihrc.com/)** – a partnership between Georgia Tech and IHRC, Inc to provide bioinformatic training and resources

* **[Abricate](https://github.com/tseemann/abricate)** – program that screens the assembled contigs for virulence genes and acquired resistance genes

* **[Apache Guacamole](https://guacamole.apache.org/)** – application that allows for remote desktop access

* **[BaseSpace](https://basespace.illumina.com/)** – user friendly workspace to manage, analyze and share sequencing data from Illumina machines

* **[BLAST](https://blast.ncbi.nlm.nih.gov/Blast.cgi)** – Basic Local Alignment Search Tool, used to identify regions of similarity between amino acid or nucleotide sequences against a database or a known sequence that are above a specific threshold

* **[CLC Genomics Workbench](https://www.qiagenbioinformatics.com/products/clc-genomics-workbench/)** – user friendly workspace for bioinformatics analysis and visualization of sequencing data, can also be used for metagenomics and RNA-seq analysis and will support sequencing data from Illumina, IonTorrent, PacBio, and GeneReader sequencing machines

* **[FastQC](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/)** – program that is used for quality control of raw reads

* **Galaxy** – web based platform for data analysis, this platform is used by Genome Trakr for sequencing analysis

* **[Illumina](https://www.illumina.com/index-d.html)** – company that provides the technology for sequencing and analysis of genomics and molecular diagnostics

* **[IqTree](http://www.iqtree.org)** – program used for phylogenetic inference using maximum likelihood with fast bootstrapping and ability to change the model used for analysis

* **[Kraken](https://github.com/DerrickWood/kraken2)** – taxonomic sequence classifier program that examines k-mers within a read and queries a database with those k-mers. This database contains a mapping of every k-mer in Kraken genomic library to the lowest common ancestor (LCA) in a taxonomic tree of all genomes that contain k-mer.

* **[Lyve – SET](https://github.com/lskatz/lyve-SET)** – (SET: Snp Extraction Tool) core genome pipeline using RaXML to examine relatedness of strains for outbreak investigation, hqSNP (high quality SNP analysis)

* **[Mash](https://github.com/marbl/Mash)** – distance estimate for genome and metagenomes

* **[Microbe TRACE (TRAnsmission Cluster Engine)](https://microbetrace.herokuapp.com/)**  – A tool for the large-scale Molecular Epidemiology of Microbial diseases and a platform for viewing and comparing genomic and contact tracing networks by the CDC. It generates network diagrams of clusters based on genetic distance data.

* **[PROKKA](https://github.com/tseemann/prokka)** – rapid prokaryotic genome annotation

* **[Quast](http://quast.sourceforge.net/quast)** – evaluates the quality of the assembled genome

* **[Raspberry Pi](https://www.raspberrypi.org/)** – a small computing chip (size of a credit card) that can be plugged into any monitor (TV or computer) and can be used to learn basic programming using Python or Scratch, commonly used in K-12 STEM education to teach computer science

* **[Roary](https://sanger-pathogens.github.io/Roary/)** – identifies the pan genome using annotated assemblies from Prokka, utilized for reference free phylogenetic analysis

* **[Seq Sero](https://github.com/denglab/SeqSero)** – Salmonella serotype identification for more than 2,300 serotypes from raw sequencing reads and genome assemblies

* **[Serotype finder](https://bitbucket.org/genomicepidemiology/serotypefinder.git)** - serotyping of Escherichia coli

* **[Shovill](https://github.com/tseemann/shovill)** – faster Spades assembler, can also run other assemblers (ie Velvet)

* **[SISTR](https://github.com/phac-nml/sistr_cmd)** – Salmonella in silico typing resource, alternative to SeqSero

* **[Spades](https://github.com/ablab/spades)** – utilized for the de novo assembly of raw reads into contigs

* **[Trimmomatic](http://www.usadellab.org/cms/?page=trimmomatic)** – read and adapter trimming of the raw reads

* **[Unicycler](https://github.com/rrwick/Unicycler)** – assembly pipeline that can assemble short read and long read only sets as well as hybrid sets that combine short and long reads

* **[Velvet](https://www.ebi.ac.uk/~zerbino/velvet/)** – package that was developed for sequencing assembly of short reads
