---
layout: default
---
## Common Bioinformatics Software

* **[StaPH-B]**(https://github.com/StaPH-B) – State public health bioinformaticians workgroup

* **[Abricate]**(https://github.com/tseemann/abricate) – program that screens the assembled contigs for virulence genes and acquired resistance genes

* **[Lyve – SET]**(https://github.com/lskatz/lyve-SET) – (SET: Snp Extraction Tool) core genome pipeline using RaXML to examine relatedness of strains for outbreak investigation, hqSNP (high quality SNP analysis)

* **[Kraken]**(https://github.com/DerrickWood/kraken2) – taxonomic sequence classifier program that examines k-mers within a read and queries a database with those k-mers. This database contains a mapping of every k-mer in Kraken genomic library to the lowest common ancestor (LCA) in a taxonomic tree of all genomes that contain k-mer.

* **[Spades]**(https://github.com/ablab/spades) – utilized for the de novo assembly of raw reads into contigs

* **[Quast]**(http://quast.sourceforge.net/quast) – evaluates the quality of the assembled genome

* **[Mash]**(https://github.com/marbl/Mash) – distance estimate for genome and metagenomes

* **[Serotype finder]**(https://bitbucket.org/genomicepidemiology/serotypefinder.git) - serotyping of Escherichia coli

* **[Seq Sero]**(https://github.com/denglab/SeqSero) – Salmonella serotype identification for more than 2,300 serotypes from raw sequencing reads and genome assemblies

* **[SISTR]**(https://github.com/phac-nml/sistr_cmd) – Salmonella in silico typing resource, alternative to SeqSero

* **[PROKKA]**(https://github.com/tseemann/prokka) – rapid prokaryotic genome annotation

* **[Trimmomatic]**(http://www.usadellab.org/cms/?page=trimmomatic) – read and adapter trimming of the raw reads

* **[IqTree]**(http://www.iqtree.org) – program used for phylogenetic inference using maximum likelihood with fast bootstrapping and ability to change the model used for analysis

* **[Unicycler]**(https://github.com/rrwick/Unicycler) – assembly pipeline that can assemble short read and long read only sets as well as hybrid sets that combine short and long reads

* **[Roary]**(https://sanger-pathogens.github.io/Roary/) – identifies the pan genome using annotated assemblies from Prokka, utilized for reference free phylogenetic analysis

* **[Shovill]**(https://github.com/tseemann/shovill) – faster Spades assembler, can also run other assemblers (ie Velvet)

* **[BLAST]**(https://blast.ncbi.nlm.nih.gov/Blast.cgi) – Basic Local Alignment Search Tool, used to identify regions of similarity between amino acid or nucleotide sequences against a database or a known sequence that are above a specific threshold

* **[Microbe TRACE] (TRAnsmission Cluster Engine)**(https://microbetrace.herokuapp.com/)  – A tool for the large-scale Molecular Epidemiology of Microbial diseases and a platform for viewing and comparing genomic and contact tracing networks by the CDC. It generates network diagrams of clusters based on genetic distance data.
